lib/charms/vault_k8s/v0/vault_autounseal.py

```diff
@@ -363,62 +392,31 @@ def get_outstanding_requests(self, relation_id: int | None = None) -> List[Relat
         credentials issued for them.
         """
         outstanding_requests: List[Relation] = []
-        requirer_requests = self.get_active_relations(relation_id=relation_id)
+        requirer_requests = self.juju_facade.get_active_relations(self.relation_name, relation_id)
         outstanding_requests = [
             relation
             for relation in requirer_requests
             if not self._credentials_issued_for_request(relation_id=relation.id)
         ]
         return outstanding_requests
 
-    def get_active_relations(self, relation_id: int | None = None) -> List[Relation]:
-        """Get all active relations on the relation name this class was initialized with.
-
-        Args:
-            relation_id: The relation ID to filter by. If None, all active relations are returned.
-
-        Returns:
-            A list of active relations.
-        """
-        relations = (
-            [
-                relation
-                for relation in self.model.relations.get(self.relation_name, [])
-                if relation.id == relation_id
-            ]
-            if relation_id is not None
-            else self.model.relations.get(self.relation_name, [])
-        )
-        return [relation for relation in relations if relation.active]
-
     def _credentials_issued_for_request(self, relation_id: int) -> bool:
-        # If Id is none and more than on relation is present we get an error
-        relation = self.model.get_relation(self.relation_name, relation_id)
-        if not relation:
+        try:
+            if not (
+                credentials_secret_id := self.juju_facade.get_app_relation_data(
+                    self.relation_name, relation_id
+                ).get("credentials_secret_id")
+            ):
+                return False
+            role_id, secret_id = self.juju_facade.get_secret_content_values(
+                "role-id",
+                "secret-id",
+                label=f"{AUTOUNSEAL_CREDENTIALS_SECRET_LABEL_PREFIX}{relation_id}",
```

Comment:
Is this a breaking change as we use the secret label now, which didn't exist for prior versions of this charm

Reply:
It shouldn't be, we don't rely on this label at all. I just thought it's nicer if the user sees a name associated with the label when running `juju secrets --format=yaml`
knowing that do you still see the risk of this being a breaking change?

Reply:
Well yes, if someone deployed Vault prior to this change, this secret wouldn't have a label. Here we are getting the secret based on the label and this method would return False, even though the secret was already created.

Reply:
Oh, I didn't mean to have it in the `get` function too. Good catch.
I left however in the `set` just to have a name for the secret.